### PR TITLE
Use django_log_formatter_ecs for ecs logging

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
- JIRA: https://issues.redhat.com/browse/RHCLOUD-11525

## Description of Intent of Change(s)
Try to enable ecs logging for RBAC.

## Local Testing
setup server locally
curl -v GET "http://0.0.0.0:8080/r/insights/platform/rbac/v1/roles/" will log to the django.server logger
`{"@timestamp": "2021-03-26T21:09:10.851154", "event": {"category": "process", "action": "rbac.middleware", "message": "{'method': 'GET', 'path': '/r/insights/platform/rbac/v1/roles/', 'status': 200, 'request_id': None, 'account': '10001', 'username': 'user_dev', 'is_admin': True, 'is_system': False}", "labels": {"application": null, "env": "rbac.settings"}, "kind": "event"}, "host": {"architecture": "x86_64", "kind": "host"}, "logline": {"level": "INFO", "kind": "logline"}}`

`{"@timestamp": "2021-03-26T21:09:10.851816", "event": {"category": "web", "action": "django.server", "message": "\"GET /r/insights/platform/rbac/v1/roles/ HTTP/1.1\" 200 4442", "labels": {"application": null, "env": "rbac.settings"}, "kind": "event"}, "host": {"architecture": "x86_64", "kind": "host"}, "logline": {"level": "INFO", "kind": "logline"}}`

wget http://127.0.0.1:8080/invalid -O /dev//null will log to both django.server and django.request loggers
`{"@timestamp": "2021-03-26T21:05:06.147886", "event": {"category": "process", "action": "rbac.middleware", "message": "{'method': 'GET', 'path': '/invalid', 'status': 404, 'request_id': None, 'account': '10001', 'username': 'user_dev', 'is_admin': True, 'is_system': False}", "labels": {"application": null, "env": "rbac.settings"}, "kind": "event"}, "host": {"architecture": "x86_64", "kind": "host"}, "logline": {"level": "INFO", "kind": "logline"}}`

`{"@timestamp": "2021-03-26T21:05:06.148856", "event": {"category": "web", "action": "django.request", "message": "Not Found: /invalid", "labels": {"application": null, "env": "rbac.settings"}, "kind": "event"}, "host": {"architecture": "x86_64", "kind": "host"}, "url": {"path": "/invalid", "domain": "127.0.0.1", "kind": "url"}, "source": {"ip": "127.0.0.1", "kind": "source"}, "httpresponse": {"status_code": "404", "kind": "httpresponse"}, "client": {"ip": "127.0.0.1", "address": "127.0.0.1", "port": 8080, "domain": "127.0.0.1", "bytes": 0, "kind": "client"}, "httprequest": {"body_content": "", "method": "GET", "body_bytes": 0, "kind": "httprequest"}, "useragent": {"original": "Wget/1.21.1", "kind": "useragent"}, "user": {"kind": "user"}, "logline": {"level": "WARNING", "kind": "logline"}}`

`{"@timestamp": "2021-03-26T21:05:06.153186", "event": {"category": "web", "action": "django.server", "message": "\"GET /invalid HTTP/1.1\" 404 77", "labels": {"application": null, "env": "rbac.settings"}, "kind": "event"}, "host": {"architecture": "x86_64", "kind": "host"}, "logline": {"level": "WARNING", "kind": "logline"}}`

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: rbac/rbac/settings.py [hunk 1]
```diff
@@ -287,11 +287,9 @@
         "ecs": {"class": "logging.StreamHandler", "formatter": "ecs_formatter"},
     },
     "loggers": {
-        "django": {"handlers": LOGGING_HANDLERS, "level": DJANGO_LOGGING_LEVEL},
-        "django.server": {"handlers": DEBUG_LOG_HANDLERS, "level": DJANGO_LOGGING_LEVEL},
-        "django.request": {"handlers": DEBUG_LOG_HANDLERS, "level": DJANGO_LOGGING_LEVEL},
```
Comment (wcmitchell) [inline rbac/rbac/settings.py:292]: So long as we're using a separate log handler here, DEBUG_LOG_HANDLERS vs LOGGING_HANDLERS these can probably stay, as the problematic data won't get passed to the ecs formatter.

File: rbac/rbac/middleware.py [hunk 2]
```diff
@@ -196,30 +195,6 @@ def process_response(self, request, response):  # pylint: disable=no-self-use
...
-                        "status_code": response.status_code,
-                    },
-                },
-                "url": {
-                    "original": request.path + query_string,
-                    "path": request.path,
-                    "query": query_string,
-                    "port": request.get_port(),
-                },
-            }
```
Comment (wcmitchell) [inline rbac/rbac/middleware.py:221]: Pulling this out leaves us lacking a lot of data we'll want for the ECS logging format, may be worth creating a future ticket to get that additional data added back in, working around the WSGIRequest and socket object problems.

**Review by wcmitchell — COMMENTED** [review]
Overall looks good, logs are getting formatted and bypassing the previously problematic logging objects. We will need to find a way to at least add in the ecs.version stanza to the emitted log messages, otherwise they don't correspond to the official schema.

Current log format:
```javascript
{
    "@timestamp": "2021-03-29T18:08:37.412180",
    "event": {
        "category": "web",
        "action": "django.server",
        "message": "\"GET /api/rbac/v1/groups/ HTTP/1.1\" 200 590",
        "labels": {
            "application": null,
            "env": "rbac.settings"
        },
        "kind": "event"
    }, "
    host": {
        "architecture": "x86_64",
    "kind": "host"
    },
    "logline": {
        "level": "INFO",
        "kind": "logline"
    }
}
```
Expected Minimal ECS Format:
```javascript
{
    "@timestamp": "2021-03-29T18:08:37.412180",
    "ecs": {
        "version": "1.6.0"
    },
    "event": {
        "category": "web",
        "action": "django.server",
        "message": "\"GET /api/rbac/v1/groups/ HTTP/1.1\" 200 590",
        "labels": {
            "application": rbac,
            "env": "rbac.settings"
        },
        "kind": "event"
    }, "
    host": {
        "architecture": "x86_64",
    "kind": "host"
    },
    "logline": {
        "level": "INFO",
        "kind": "logline"
    }
}
```
Expected Full ECS Format:
```javascript

{
    "@timestamp": "2021-03-29T18:08:37.412180",
    "ecs": {
        "version": "1.6.0"
    },
    "event": {
        "category": "web",
        "action": "django.server",
        "message": "\"GET /api/rbac/v1/groups/ HTTP/1.1\" 200 590",
        "labels": {
            "application": rbac,
            "env": "rbac.settings"
        },
        "kind": "event"
    }, "
    host": {
        "architecture": "x86_64",
    "kind": "host"
    },
    "http": {
        "request": {
            "body": {
                "bytes": 0
            },
            "bytes": 5,
            "method": GET
        },
        "response": {
            "body": {
                "bytes": 15
            },
            "bytes": 20,
            "status_code": 200,
        },
    },
    "url": {
        "original":  /api/rbac/v1/groups/,
        "path":  /api/rbac/v1/groups/,
        "query": "",
        "port": 8000,
    },
    "logline": {
        "level": "INFO",
        "kind": "logline"
    }
}
```

**Comment by astrozzc** [comment]
Abandon this django_log_formatter_ecs https://github.com/RedHatInsights/insights-rbac/pull/463

**Pull request closed without merging**